src/cart/crt_group.c

```diff
@@ -1750,9 +1752,16 @@ open_tmp_attach_info_file(char **filename)
 		return NULL;
 	D_ASSERT(*filename != NULL);
 
+	/** Ensure the temporary file is created with proper permissions to
+	 *  limit security risk.
+	 */
+	old_mode = umask(S_IRUSR | S_IWUSR);
```

Comment:
in case if server runs as root and client runs as user, we probably want this to be  | S_IRGRP | S_IROTH